go/pserver/optimizer.go

```diff
@@ -78,7 +80,11 @@ func (o *optimizer) UpdateParameter(g Gradient) error {
 		return fmt.Errorf("Name: %s, parameter and gradient element type not match, parameter: %v, gradient: %v", g.Name, o.elementType, g.ElementType)
 	}
 
-	r := C.paddle_update_parameter(o.opt, C.paddle_element_type(g.ElementType), unsafe.Pointer(&g.Content[0]), C.int(len(g.Content))/C.sizeof_float)
+	if o.contentLen != len(g.Content) {
+		return fmt.Errorf("Name: %s, parameter and gradient does not have same content len, parameter: %d, gradient: %d", g.Name, o.contentLen, len(g.Content))
+	}
+
+	r := C.paddle_update_parameter(o.opt, C.paddle_element_type(g.ElementType), unsafe.Pointer(&g.Content[0]), C.int(len(g.Content)))
```

Comment:
`num_bytes` is for `const void* grad_buffer` which means the length of the buffer. Then `paddle_update_parameter` need to calculate the array_size based of the type of the element. That's why I changed from `new TensorT(buffer, num_bytes)` to `new Tensor(buffer, num_bytes/sizeof(float))` [here](https://github.com/PaddlePaddle/Paddle/pull/2855/files#diff-defc4d95efae04a339f807e16eb7265aR47)